src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -270,16 +250,35 @@ private boolean isPrivateForNotification(UUID parentDirectoryUuid, boolean isCur
             ? Mono.empty() : Mono.error(new DirectoryException(NOT_FOUND));
     }
 
-    public Flux<ElementAttributes> getElementsAttribute(List<UUID> ids) {
-        return Flux.fromStream(() -> directoryElementRepository.findAllById(ids).stream().map(ElementAttributes::toElementAttributes));
+    public Flux<ElementAttributes> getElements(List<UUID> ids) {
+        return Mono.fromCallable(() -> directoryElementRepository.findAllById(ids))
+            .flatMapMany(elementEntities ->
+                elementEntities.size() != ids.size() ?
+                    Flux.error(new DirectoryException(NOT_FOUND)) :
+                    Flux.fromStream(elementEntities.stream().map(ElementAttributes::toElementAttributes)));
+    }
+
+    public Mono<Void> notify(@NonNull String notificationName, @NonNull UUID elementUuid, @NonNull String userId) {
+        NotificationType notification;
+        try {
+            notification = NotificationType.valueOf(notificationName.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            return Mono.error(DirectoryException.createNotificationUnknown(notificationName));
+        }
+
+        if (notification == UPDATE_DIRECTORY) {
+            return emitDirectoryChangedNotification(elementUuid, userId);
+        } else {
+            return Mono.error(DirectoryException.createNotificationUnknown(notification.name()));
+        }
     }
 
     public Mono<Void> emitDirectoryChangedNotification(UUID elementUuid, String userId) {
-        return getElementInfos(elementUuid).flatMap(elementAttributes -> {
-            UUID parentUuid = getParentUuid(elementUuid);
-            emitDirectoryChanged(parentUuid, userId, elementAttributes.getAccessRights().isPrivate(), parentUuid == null,
-                NotificationType.UPDATE_DIRECTORY);
-            return Mono.empty();
-        });
+        return getElement(elementUuid)
+            .doOnNext(elementAttributes -> {
+                UUID parentUuid = getParentUuid(elementUuid);
+                emitDirectoryChanged(parentUuid, userId, elementAttributes.getAccessRights().isPrivate(), parentUuid == null, UPDATE_DIRECTORY);
```

Comment:
See previous comment about use of NotificationType.UPDATE_DIRECTORY

Reply:
Done